cmd/jag/commands/run.go

```diff
@@ -5,20 +5,48 @@
 package commands
 
 import (
-	"errors"
+	"bytes"
 	"fmt"
 	"io"
 	"io/ioutil"
 	"os"
 	"path/filepath"
-	"strings"
 
-	"github.com/blakesmith/ar"
+	"github.com/setanta314/ar"
 	"github.com/google/uuid"
 	"github.com/spf13/cobra"
 	"github.com/toitlang/jaguar/cmd/jag/directory"
 )
 
+// Checks whether a file is a snapshot file.  Starts by checking for an ar
+// file, since snapshot files are ar files.
+func IsSnapshot(filename string) bool {
+	file, err := os.Open(filename)
+	if err != nil {
+		return false
+	}
+	defer file.Close()
+	magic_sequence := make([]byte, 8)
+	_, err = io.ReadAtLeast(file, magic_sequence, 8)
+	if err != nil {
+		return false
+	}
+	if bytes.Compare(magic_sequence, []byte("!<arch>\n")) != 0 {
+		return false
+	}
+
+	file.Seek(0, io.SeekStart)
+	reader := ar.NewReader(file)
+	header, err := reader.Next()
+	if err != nil {
+		return false
+	}
+	if header.Name != "toit" {
+		return false
+	}
+	return true
```

Comment:
You could add a `reader.Read` to check that the content is correct, but it's probably really not necessary.